content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,160 @@ title="Specify the environment"
+* Shared libraries, such as `Some-shared-library` and `some-static-library`, are added to `LD_LIBRARY_PATH` and `LIBRARY_PATH`
+* If `env/SOME_VAR` and `env.build/SOME_VAR` have a conflict, a given procedure applies to figure out who "wins" when applying the environment [modification rules](#environment-variable-modification-rules).
+* `User-defined` variables are then applied; meaning it's possible to override `buildpack-defined` variables.
+* Finally, `platform-defined` variables are applied, which eventually override any previous values.
+
+The above is only relevant for `build` because the layer `some-build-layer` is a build layer, i.e., `some-build-layer.toml` has
+
+```yaml
+
+[types]
+build = true
+
+```
+
+>Note that the platform SHOULD NOT assume any other `base-image` provided environment variables are inherited by the buildpack.
+
+### Build Environment Variables
+
+During the `build` phase, buildpacks MAY write environment variable files to `<layers>/<layer>/env/`, `<layers>/<layer>/env.build/`, and `<layers>/<layer>/env.launch/` directories.
+
+For each `<layers>/<layer>/` designated as a build layer, for each file written to `<layers>/<layer>/env/` or `<layers>/<layer>/env.build/` by `/bin/build`, the `lifecycle` MUST modify an environment variable in subsequent executions of `/bin/build` according to the [environment variable modification rules](https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules).
+
+### Process and Runtime Environment Variables
+
+For `runtime` and `process` environment variables, the [tree above](#example) is still applicable except that it doesn't include a `build` layer. Instead it has a`launch` layer, i.e., `some-launch-layer.toml` that has
+
+```yaml
+
+[types]
+launch = true
+
+```
+
+### Multiple Buildpacks Defining the Same Environment Variable
+
+During the `build` phase, each variable designated for `build` MUST contain absolute paths of all previous buildpacks’ `<layers>/<layer>/` directories that are designated for build.
+
+When the exported `OCI image` is launched, each variable designated for launch MUST contain absolute paths of all buildpacks’ `<layers>/<layer>/` directories that are designated for launch.
+
+In either case,
+
+* The lifecycle MUST order all `<layer>` paths to reflect the reversed order of the buildpack group.
+* The lifecycle MUST order all `<layer>` paths provided by a given buildpack alphabetically ascending.
+* The lifecycle MUST separate each path with the OS path list separator (e.g. `:` on Linux, `;` on Windows).
+
+#### Environment Variable Modification Rules
+
+The lifecycle MUST consider the name of the environment variable to be the name of the file up to the first period (`.`) or to the end of the name if no periods are present. In all cases, file contents MUST NOT be evaluated by a shell or otherwise modified before inclusion in environment variable values.
```

Comment:
This is as far as I got in my review. I think we can still re-work this section to be more for a buildpack author (what they can expect to find in their environment when a previous buildpack does something) vs describing the lifecycle's behavior.